stdlib/src/python/_cpython.mojo

```diff
@@ -288,8 +288,16 @@ struct PythonVersion:
         self = PythonVersion(components[0], components[1], components[2])
 
 
-fn _py_get_version(lib: DLHandle) -> StringRef:
-    return StringRef(ptr=lib.call["Py_GetVersion", UnsafePointer[c_char]]())
+fn _py_get_version(lib: DLHandle) -> StringSlice[ImmutableAnyOrigin]:
+    var ptr = lib.call["Py_GetVersion", UnsafePointer[c_char]]()
+    var length = 0
+    for i in range(50):
+        if ptr[i] == 0:
+            length = i
+            break
+    return StringSlice[ImmutableAnyOrigin](
+        ptr=ptr.bitcast[Byte](), length=length
+    )
```

Comment:
I'd rather implement an unbounded version here if it's the limit what worries you. I want to fully deprecate `StringRef` ASAP and that means stop depending on it. I figured no Python version string is ever going to exceed 50 bytes :man_shrugging:

Reply:
I think you're on the right track with trying to eliminate `StringRef` 🙂 , but I think having a Mojo function that implements C `strlen` is useful, and we should avoid reimplementing `strlen` via for loop in each place its needed. Having a dedicated function for this logic makes the code locally more readable, and makes it easier to spot places where we've got logic like this that could be cleaned up in future refactorings.

Since we aren't able to entirely eliminate `StringRef` in this PR anyway, what do you think about continuing to use `StringRef.__init__(UnsafePointer[c_char])` for now as `strlen()` effectively, and then in a follow-up PR we can factor that out into a free function (or maybe even use it as part of a `CString` type in the standard library 🙂)?

(Re. your [comment here](https://github.com/modularml/mojo/pull/3775#discussion_r1854355245), I think I'd prefer we not add `strlen()` in this PR, as this is already a large change set, which can make review difficult.)

I think that will hill-climb us towards the goal of using `StringRef` less widely, while laying a clear path for future cleanup work 🙂